src/Bicep.Cli/Services/CompilationService.cs

```diff
@@ -29,14 +30,16 @@ public class CompilationService
         private readonly InvocationContext invocationContext;
         private readonly Workspace workspace;
         private readonly TemplateDecompiler decompiler;
+        private readonly ApiVersionProvider apiVersionProvider;
```

Comment:
Is there a way we can avoid passing this as a parameter everywhere?

Reply:
@anthony-c-martin Yeah, that's annoying.  Most of the annoyance comes from this kind of repeated code:

            var compilation = new Compilation(BicepTestConstants.Features, TestTypeHelper.CreateEmptyProvider(), SourceFileGroupingFactory.CreateFromText(program, BicepTestConstants.FileResolver), BicepTestConstants.BuiltInConfiguration, BicepTestConstants.ApiVersionProvider, BicepTestConstants.LinterAnalyzer);

Notice that the args list for Compilation's constructor is slowly growing.  I could make lots of that params optional and use defaults if left out, but that doesn't really seem like a good solution.

Easiest fix is simply to add a helper method for "new Compilation" that's used only in tests and uses defaults with optional overrides, e.g. CreateTestCompilation().  Sound good?

Reply:
Great suggestion - that should help with a lot of the test duplication!

Reply:
Will do after merge.